clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1388,9 +1388,9 @@ public void commitSync() {
      */
     @Override
     public void commitSync(Duration timeout) {
+        maybeThrowInvalidGroupIdException();
```

Comment:
If we throw here, we will not execute the current `finally` block to call `release()`